tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -102,21 +103,59 @@ class AppFilterImpl(
       } else {
         appNameFiltered
       }
-      appNamelogicFiltered.filter(_.appInfo.exists(_.userName.contains(userName)))
+      appNamelogicFiltered.filter(_.appInfo.appStartInfo.exists(_.userName.contains(userName)))
     } else {
       appNameFiltered
     }
 
+    val configFiltered = if (configurationNames.nonEmpty) {
+      val userNameLogicFiltered = if (appArgs.any()) {
+        apps
+      } else {
+        userNameFiltered
+      }
+
+      if (configurationNames.toString.contains(":")) { // check for specific configuration values
```

Comment:
so if any configs have values?  What if you mix and some have values and some don't?

Reply:
If it was mix of key:value pairs and only keys, I was ignoring the keys configs. Had documented the behavior.
But I guess I should be considering both. i.e we may want to filter some with key:value pairs and some with only keys in the same command.